src/compiler/base/initial-check/perform-initial-check.xsl

```diff
@@ -28,7 +28,7 @@
             </xsl:when>
             <xsl:otherwise>
                <!-- Always write a single non-empty line to help Bats tests to predict line numbers. -->
-               <xsl:text> </xsl:text>
+               <xsl:text>Checking for deprecated Saxon versions: Passed</xsl:text>
```

Comment:
I'm open to other ideas for this text. The space character causes many bats tests to fail with Saxon 11 because the line numbering is off by one. A nonbreaking space looks ugly in a shell window. This text adds clutter to our test logs, but at least that's not user-facing.

Reply:
@AirQuick : Is this the part of this PR where you [wanted to explore](https://github.com/xspec/xspec/pull/1719#issuecomment-1536549244) alternatives, such as the method in #1719? Maybe I can also explore alternatives, if I know more about what your concerns are. Are you thinking...

a) That a PR that centers on testing should avoid making a user-visible change?
b) That a PR that centers on testing should avoid touching compiler files -- and if touching compiler code _is_ necessary, the modification should be located in the test/ area only?
c) Something else?

If it's only (a), then maybe the single-line message can be emitted only during automated testing. Doing that by changing `<xsl:otherwise>` here to something like `<xsl:when test="environment-variable('XSPEC_TEST_ENV') != ''">` would be simpler than the method in #1719 but it would not address (b).

Reply:
According to https://saxonica.plan.io/issues/6041, there is no instant solution for this.
I think this PR in its current form is on a good balance between usability and testability.

> Is this the part of this PR where you [wanted to explore](https://github.com/xspec/xspec/pull/1719#issuecomment-1536549244) alternatives, such as the method in #1719?

No. My comment in #1719 was just for the issues around the _skeleton_ Schematron implementation.